pkg/formulaexec/formula_exec.go

```diff
@@ -504,7 +506,9 @@ func invokeRunc(ctx context.Context, config runConfig, logWriter io.Writer) (str
 		fmt.Sprintf("warpforge-%d", time.Now().UTC().UnixNano()), // container id
 	)
 
-	if config.spec.Process.Terminal {
+	// if the config has terminal enabled, and interactivity is requested,
+	// wire stdin to the contaniner
+	if config.spec.Process.Terminal && config.interactive {
 		cmd.Stdin = os.Stdin
```

Comment:
Should we be passing an `io.Reader` instead of using `os.Stdin` directly?

Reply:
Absolutely not.  The `os.Stdin` handle "happens to be" also a device file which is imbued with all the tty wondersauces.  Passing a golang level buffer around would cause drastically different behaviors.

(... usually, of course.  If you were to pipe things into the warpforge process, then of course stdin already isn't a tty.  IIUC, that's part of what the whole `config.spec.Process.Terminal` reference is handling.)